File: AIO/AIO 2024/[p3] Shopping Spree.cpp
#include <bits/stdc++.h>
using namespace std;
 
int main() {
    int N, K;
    cin >> N >> K;
    vector<int> C(N);
    for(int i = 0; i < N; i++) cin >> C[i];
    int discounted = 0;
    for(int i = 0; i < K; i++) {
        discounted += C[i];
    }
    int other = 0;
    for(int i = K + 1; i < N - K; i += 2) {
        other += C[i];
    }
 
    cout << (discounted + other);
}

File: AIO/AIO 2024/[p3] Shopping Spree.py
input_file = open('shopin.txt', 'r')
N, K = map(int, input_file.readline().split())
C = list(map(int, input_file.readline().split()))
 
discounted = sum(C[:K])
other = sum(C[K+1:N-K:2])
 
output_file = open('shopout.txt', 'w')
output_file.write(str(discounted+other))
output_file.close()
